File: src/main/java/de/emaarco/example/adapter/outbound/InMemoryTaskRepository.java
package de.emaarco.example.adapter.outbound;

import de.emaarco.example.application.port.outbound.LoadTasksPort;
import de.emaarco.example.application.port.outbound.SaveTaskPort;
import de.emaarco.example.domain.Task;
import jakarta.enterprise.context.ApplicationScoped;
import java.util.ArrayList;
import java.util.List;

@ApplicationScoped
public class InMemoryTaskRepository implements SaveTaskPort, LoadTasksPort {

    private final List<Task> taskList = new ArrayList<>();

    @Override
    public List<Task> loadTasks() {
        return taskList;
    }

    @Override
    public void saveTask(Task task) {
        taskList.add(task);
    }
}

File: src/main/kotlin/de/emaarco/example/adapter/outbound/InMemoryTaskRepository.kt
package de.emaarco.example.adapter.outbound

import de.emaarco.example.application.port.outbound.LoadTasksPort
import de.emaarco.example.application.port.outbound.SaveTaskPort
import de.emaarco.example.domain.Task
import jakarta.enterprise.context.ApplicationScoped

@ApplicationScoped
class InMemoryTaskRepository : SaveTaskPort, LoadTasksPort {

    private val taskList = mutableListOf<Task>()

    override fun loadTasks(): List<Task> = taskList

    override fun saveTask(task: Task) {
        taskList.add(task)
    }
} 